enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthScenariosInteractionTestBase.java

```diff
@@ -572,7 +571,8 @@ public void rolesListing() throws Throwable
     Admin lists all roles for user Craig → ok
     Admin adds user Henrik to role Publisher
     Craig logs in with correct password → ok
-    Craig lists all roles for user Craig → ok
+    Craig lists all roles for user Craig → permission denied
```

Comment:
I'm pretty sure you cannot change this behavior, that would be a functionality regression. In general I don't think we can change the semantics of any of these tests until `4.0`.

Reply:
If we can't change this behaviour, then the two procedures that suffer from this (`changeUserPassword` & `listRolesForUser`) cannot use the new annotation.

Though we could also argue that this is a bug fix, since both the manual and `db.procedures` specify that these are only executable by `admin`. There's nothing about being allowed to execute them with your own username.

This is also the reason for the removed tests, part of the tests are merged with other tests and part is removed due to not being applicable any longer and replaced with other tests.

Reply:
Then is think we just shouldn't use the annotation for these two procedures. The cost of keeping the behavior is very small, so why risk breaking backwards compatibility.

Yelling bug would be dishonest, we explicitly added _acceptance tests_ to ensure that this functionality was provided. We clearly intended it to work that way :).